Map/Source/SquareMap.cpp

```diff
@@ -62,4 +63,133 @@ SquareMap::SquareMap(const int mapSize)
     {
         _fieldList.at(i).at(0) = std::make_unique<FightField>();
     }
-}
\ No newline at end of file
+}
+
+SquareMap::SquareMap(FieldList&& fieldList) : _fieldList(std::move(fieldList))
+{
+}
+
+bool SquareMap::isFieldAccessible(const Position& position)
+{
+    if (not isField(position) or getField(position)->getType() == FieldType::Wall)
+    {
+        return false;
+    }
+    return true;
+}
+
+bool SquareMap::isField(const Position& position)
+{
+    try
+    {
+        if (getField(position))
+        {
+            return true;
+        }
+        return false;
+    }
+    catch (std::out_of_range&)
+    {
+        return false;
+    }
+}
+
+const std::unique_ptr<Field>& SquareMap::getField(const Position& position)
+{
```

Comment:
i ten try/catch powinien byc tutaj, zeby kazde uzycie getField było "pilnowane". Bo wyzej masz to ogarniete, ale w updateVisilibity juz nie, i wywali nam program jesli podamy zle pozycje

Reply:
Tylko co ma zwracać przy catchu? null pointer?